### PR TITLE
brew recycler core adapter

### PR DESCRIPTION
**Pull request by roideuniverse**

### CODE REVIEW

File: recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ClassUtil.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Created by kaushik on 9/30/16.
+ */
+
+class ClassUtil
+{
+    static String getClassName(TypeElement type, String packageName)
+    {
+        int packageLen = packageName.length() + 1;
+        return type.getQualifiedName().toString().substring(packageLen).replace('.', '$');
```
Comment (BrandonRomano) [inline recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ClassUtil.java]: Why `.replace('.', '$');` here?

**Comment by BrandonRomano** [comment]
Super solid/clean man 👍 👍

File: recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ClassUtil.java (hunk 1 above)
Comment (roideuniverse) [reply recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ClassUtil.java]: Good catch, I believe it was for Inner classes. For example 

```
package app.ui
MainActivity {
  @RCModel
  TestModel {
  }
}
```

In this case the package would be `app.ui` and the class will be `MainActivity.TestModel` which this util function would change it into `MainActivity$TestActivity`

I just tried it and hit a roadblock, will need to test this and get back to you.

Comment (BrandonRomano) [reply recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ClassUtil.java]: Sure thing... I'll assign this PR back to you until then!

**Review by BrandonRomano — APPROVED** [review]

File: recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/PreProcessorHelper.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+/**
+ * Created by kaushiksaurabh on 9/19/16.
+ */
+
+class PreProcessorHelper
+{
+    private Elements mElements;
+    private Types mTypes;
+    private Filer mFiler;
```
Comment (asolis90) [inline recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/PreProcessorHelper.java]: mFiler and mElements are not being used here

File: recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ModelDetails.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        ClassName clsName = ClassName.get(typeElement);
+        String cannonName = ClassUtil.getCanonicalName(clsName);
+        String controllerCanonicalName = null;
+        try
+        {
+            RCModel annotation = typeElement.getAnnotation(RCModel.class);
+            annotation.controller();
+        }
+        catch(MirroredTypeException ex)
```
Comment (asolis90) [inline recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ModelDetails.java:74]: Could you add a comment here that we are getting the controller's canonical name in the MirroredException

**Review by asolis90 — CHANGES_REQUESTED** [review]
Looks good to merge after fixing some minor comments

File: recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ModelDetails.java (hunk 3 above)
Comment (roideuniverse) [reply recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ModelDetails.java:74]: Added as method comment.

**Review by asolis90 — APPROVED** [review]

**Comment by roideuniverse** [comment]
I no longer have access to merge this. Please merge.

**Comment by tinder-ksaurabh** [comment]
@BrandonRomano @asolis90 Guys merge this PR!

**Comment by BrandonRomano** [comment]
Sorry about that @tinder-ksaurabh -- forgot to merge this in.. Doing it now 🚀

**Comment by roideuniverse** [comment]
Thanks @BrandonRomano :+1:

**Comment by BrandonRomano** [comment]
You got it 👌 sorry it took so long 😬 .

Will get you access to this repo Monday.

**Pull request merged**